src/polytopes/segment.jl

```diff
@@ -36,15 +36,16 @@ end
 
 function Base.in(p::Point{Dim,T}, s::Segment{Dim,T}) where {Dim,T}
   a, b = s.vertices
-  if Dim == 2
-    _0 = zero(T)
-  elseif Dim == 3
-    _0 = Vec{3,T}(0, 0, 0)
+  ab, ap = b - a, p - a
+  # points a, b, p are collinear if and only if the cross-products (cp) for ab and ap
+  # with respect to all possible pairs of coordinates are zero
+  if T == 1
+    iscollinear = true
   else
-    throw(ErrorException("not implemented"))
+    cp = [Vec{2,T}(ab[[i, j]]) × Vec{2,T}(ap[[i, j]]) for i in 1:Dim for j in (i+1):Dim]
```

Comment:
We should probably split this into a for loop with premature termination whenever a pair fails the condition? Right now you are computing all possible cross-products to then check later if they are all zero. You could write a simple for loop and break the loop if any intermediate result is non-zero. Makes sense? This can also improve readibility.

I wonder if you have a reference to support this statement that we could link here in the comments for future readers?

Reply:
@ErickChacon can you refactor this into a for loop with early termination? You can do something like:

```julia
iscollinear = true
for i in 1:Dim, j in (i+1):Dim
  # do cross product with i and j
  if !iszero(cp)
    iscollinear = false
    break
  end
end
```